Game.java

```diff
@@ -0,0 +1,250 @@
+			Ydim = topBoard.get()[0].length;
+		}
+		
+		nextLocation.setX( setDimensionOutOfBoard( nextLocation.getX(), Xdim ) );
+		nextLocation.setY( setDimensionOutOfBoard( nextLocation.getY(), Ydim ) );
+		
+		return nextLocation;
+	}
+
+	private int setDimensionOutOfBoard(int location, int dimension) {
+		// This method relies on the equality of the board dimensions
+		if (location >= dimension)
+			return 0;
+		
+		if (location < 0)
+			return dimension - 1;
+		
+		return location;
+	}
+	
+	private void moveGhost(Board.Creature c){
+		
+		Location currentLocation = c.getLocation();
+		Location nextLocation = getNextLocation(c, c.getDirection());
+		
+		switch ( getCreatureAt(nextLocation)){
+			case Point:
+			case Null:
+				topBoard.set( currentLocation, Board.Creature.Null );
+				topBoard.set( nextLocation, c );
+				break;
+			case Pacman:
+				stopGame( c );
+				break;
+
+			default:
+				break;					
+		}
+		
+		
+	}
+
+	private Board.Creature getGhostCreatureByNum(int num){
+		switch (num){
+			case 1:
+				return Board.Creature.Ghost1;
+			case 2:
+				return Board.Creature.Ghost2;
+			case 3:
+				return Board.Creature.Ghost3;
+			case 4:
+				return Board.Creature.Ghost4;
+				
+			default:
+				return null; // might cause problems
+		}
+	}
+	
+	private Location changeLocationByDirection(int x, int y, Direction d){
+		switch (d){
+		case Right:
+			y ++;
+			break;
+		case Left:
+			y --;
+			break;
+		case Down:
+			x ++;
+			break;
+		case Up:
+			x --;
+			break;
+		}
+		return new Location(x, y);
+	}
+	
+	private void stopGame(Board.Creature gameStopper){
+		// in actual game, I will stop the game using this function
+		
+		System.out.print("Game Stopped - ");
+		if (gameStopper == Board.Creature.Pacman)
+			System.out.println("Pacman Commited Suicide");
+		else
+			System.out.println(gameStopper.toString() + " Ate Pacman");
+		
+		this.moveTimer.cancel();
+		this.gameOn = false;
+	}
+	
+	private Vector<Direction> findNewPathForGhost(Board.Creature ghost){
+		Vector<Direction> possibleDirections = new Vector<>(0, 1);
+		
+		Direction previousDirection = ghost.getDirection();
+		
+		for (Direction d : Direction.values()){
+			if ( getCreatureAt( getNextLocation(ghost, d) ) != Board.Creature.Wall &&
+					d != previousDirection)
+				possibleDirections.add(d);
+		}
+		
+		if (possibleDirections.capacity() == 0)
+			possibleDirections.add( getCreatureAt( getNextLocation( ghost, previousDirection ) )
+					!= Board.Creature.Wall ? previousDirection : defaultDirection);
+		
+		return possibleDirections;
+		
+	}
+	
+	public Board.Creature getCreatureAt(Location l){
+		
+		// If the creature is null on the top, it will return the creature on the bottom board
+		// Otherwise method returns pacman / ghost from top board
+		
+		Board.Creature topBoardCreature = this.topBoard.get(l);
+		
+		if (topBoardCreature == Board.Creature.Null) 
+			return this.botBoard.get(l);
+		else
+			return topBoardCreature;
+	}
+	
+	public void move(){
```

Comment:
I think that implementing it as a list of "Creature" object and then call each one "move" method is better than knowing the parts of the game.
The logic of ghost movement should be kept inside the ghost class, as well as pacman movement (for now it's something you gonna ask from the user, but maybe you can implement it as a super-AI agent in the future??)
Consider it as a good refactor sometime in the near future